src/test/kotlin/lotto/domain/LottoTicketIssuerTest.kt

```diff
@@ -0,0 +1,38 @@
+package lotto.domain
+
+import io.kotest.assertions.throwables.shouldThrowWithMessage
+import io.kotest.matchers.shouldBe
+import lotto.domain.LottoTicketIssuer.DEFAULT_LOTTO_PRICE
+import lotto.domain.LottoTicketIssuer.INVALID_MIN_COST_LOTTO_PAID_MESSAGE
+import lotto.domain.LottoTicketIssuer.INVALID_THOUSAND_UNIT_LOTTO_PAID_MESSAGE
+import lotto.view.lottoNumberGenerator
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+class LottoTicketIssuerTest {
+    @ParameterizedTest
+    @ValueSource(ints = [500, 999])
+    fun `로또 구입 비용이 1,000원 미만일 경우 에러가 발생한다`(amountPaid: Int) {
+        shouldThrowWithMessage<IllegalArgumentException>(message = INVALID_MIN_COST_LOTTO_PAID_MESSAGE) {
+            LottoTicketIssuer.issueTickets(amountPaid = amountPaid, lottoNumberGenerator = { lottoNumberGenerator() })
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = [1500, 22200, 10500, 2100005])
+    fun `로또 구입 비용이 1,000원 단위가 아닐 경우 에러가 발생한다`(amountPaid: Int) {
+        shouldThrowWithMessage<IllegalArgumentException>(message = INVALID_THOUSAND_UNIT_LOTTO_PAID_MESSAGE) {
+            LottoTicketIssuer.issueTickets(amountPaid = amountPaid, lottoNumberGenerator = { lottoNumberGenerator() })
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = [5000, 1000, 23000, 3000])
+    fun `로또 구입 비용에 대해 몇 장의 로또가 구입됐는지 확인활 수 있다`(amountPaid: Int) {
+        val purchasedLottoTickets =
+            LottoTicketIssuer.issueTickets(amountPaid = amountPaid, lottoNumberGenerator = { lottoNumberGenerator() })
+        val purchasedCount = amountPaid / DEFAULT_LOTTO_PRICE
+        purchasedLottoTickets.purchasedCount shouldBe purchasedCount
+        purchasedLottoTickets.purchasedLottoTickets.size shouldBe purchasedCount
+    }
```

Comment:
테스트는 직관적일수록 좋다고 생각합니다.  `@CsvSource`를 활용하면 아래와 같이 결과값도 함께 넣어주어 테스트가 가능할 것 같아요!

```suggestion
    @ParameterizedTest
    @CsvSource("5000,5", "1000,1")
    fun `로또 구입 비용에 대해 몇 장의 로또가 구입됐는지 확인활 수 있다`(amountPaid: Int, expectedTicketCount: Int) {
        val purchasedLottoTickets =
            LottoTicketIssuer.issueTickets(amountPaid = amountPaid, lottoNumberGenerator = { lottoNumberGenerator() })
        purchasedLottoTickets.purchasedCount shouldBe expectedTicketCount
        purchasedLottoTickets.purchasedLottoTickets.size shouldBe expectedTicketCount
    }
```

+) 
지금 작성해주신 테스트 코드는 `DEFAULT_LOTTO_PRICE`가 변경되더라도 성공하는 유연한 테스트라고 볼 수 있지만, 반대로 내부 구현을 알아야만 작성될 수 있는 테스트코드라서 장단이 있다고 생각해요 :) 이에 대한 생각도 궁금합니다!

Reply:
좋은 피드백 감사합니다!

말씀해주신 @CsvSource를 활용해서 테스트 케이스를 명확하게 하는 것이 더 직관적이기 때문에 코드를 이해하기 쉬워지는 좋은 방법이라고 생각합니다 👍 

`DEFAULT_LOTTO_PRICE`가 변경되더라도 성공하는 유연한 테스트의 장단점에 대한 의견을 주신 부분은, 개인적으로도 여러 생각을 하게 되는 부분 같습니다

실제로, `DEFAULT_LOTTO_PRICE`가 변경되더라도 성공하는 테스트는 리팩토링할 때 유리한 점이 많겠지만, 
내부 구현을 알아야만 작성될 수 있는 코드가 맞습니다.

궁극적으로 테스트 코드는 명확한 요구사항을 표현하고, 프로덕션 코드의 기대 동작을 검증하는 것에 있음으로 
내부 구현을 알아야만하는 테스트 코드는 지양하는게 올바른 방향이지 않을까 생각하게 되었습니다.

피드백 주신 부분들을 반영하여 테스트 코드를 개선해 보도록 하겠습니다. 감사합니다!